src/scripts/main.js

```diff
@@ -1,3 +1,286 @@
+      arrowMove(link);
+      break;
+    case 'ArrowLeft':
+      arrowMove(link, true);
+      break;
+    case 'ArrowDown':
+      arrowMove(rotating, false, true);
+      break;
+    case 'ArrowUp':
+      arrowMove(rotating, true, true);
+      break;
+  }
+
+  checkOnWin(table2);
+
+  for (const row of table2) {
+    if (!row.includes(0)) {
+      const checkTable = deepClone(table2);
+      const down = transpose(moveCells(rotating, true));
+      const up = transpose(moveCells(rotating, false));
+      const right = moveCells(table2, false);
+      const left = moveCells(table2, true);
+      const results = [
+        onChanges(checkTable, down),
+        onChanges(checkTable, up),
+        onChanges(checkTable, right),
+        onChanges(checkTable, left),
+      ];
+
+      if (results.every(el => el === false)) {
+        messageLose.classList.remove('hidden');
+        starting = false;
+      }
+    }
+  }
+});
+
+function transpose(original) {
+  const copy = [];
+
+  for (let i = 0; i < original.length; ++i) {
+    for (let j = 0; j < original[i].length; ++j) {
+      if (original[i][j] === undefined) {
+        continue;
+      }
+
+      if (copy[j] === undefined) {
+        copy[j] = [];
+      }
+
+      copy[j][i] = original[i][j];
+    }
+  }
+
+  return copy;
+}
+
+function moveCells(matrix, reverse = false, original = false) {
+  const copy = !(reverse)
+    ? deepClone(matrix).map(elem => elem.reverse())
+    : deepClone(matrix);
+
+  function removeZeroes(row) {
+    return row.filter(num => num !== 0);
+  }
+
+  function slide(row) {
+    let newRow = removeZeroes(row);
+
+    for (let i = 0; i < newRow.length - 1; i++) {
+      if (newRow[i] === newRow[i + 1]) {
+        newRow[i] *= 2;
+        newRow[i + 1] = 0;
+
+        if (original) {
+          addScore(newRow[i]);
+        }
+      }
+      newRow = removeZeroes(newRow);
+    }
+
+    while (newRow.length < column) {
+      newRow.push(0);
+    }
+
+    return newRow;
+  }
+
+  const copy1 = [];
+
+  for (let j = 0; j < rows.length; j++) {
+    let row = copy[j];
+
+    row = slide(row);
+    copy1[j] = row;
+  }
+
+  return (!(reverse) ? copy1.map(elem => elem.reverse()) : copy1);
+}
+
+function addScore(plusNumber) {
+  const oldScores = +score.textContent;
+  const newScores = oldScores + plusNumber;
+
+  score.textContent = newScores;
+}
+
+function addValuesToCells(matrix) {
+  for (let i = 0; i < rows.length; i++) {
+    for (let j = 0; j < rows.length; j++) {
+      rows[i].cells[j].textContent = (matrix[i][j] !== 0)
+        ? matrix[i][j]
+        : ' ';
+    }
+  }
+}
+
+function addInRandomCell(matrix, percent = 10) {
+  const emptyCellsIndexes = [];
+  const randomNumber = Math.round(Math.random() * 100);
+  const value = randomNumber > percent
+    ? 2
+    : 4;
+
+  for (let i = 0; i < rows.length; i++) {
+    for (let j = 0; j < rows.length; j++) {
+      if (matrix[i][j] === 0) {
+        emptyCellsIndexes.push(`${i}${j}`);
```

Comment:
```suggestion
        emptyCellsIndexes.push([i, j]);
```
as an option you could do it like this